### PR TITLE
refactor: layer it on top of rules_js

### PR DESCRIPTION
**Pull request by thesayyn**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -2,57 +2,58 @@ name: CI
...
+            matrix:
+                folder:
+                    - '.'
+                    - 'e2e/workspace'
+                    - 'e2e/worker'
+
+        # Steps represent a sequence of tasks that will be executed as part of the job
+        steps:
+            # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+            - uses: actions/checkout@v2
```
Comment (alexeagle) [inline .github/workflows/ci.yaml:29]: nit: this one and actions/cache below should be v3 now or renovate will update

File: .github/workflows/workspace_snippet.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# Strip leading 'v'
+PREFIX="rules_ts-${TAG:1}"
+SHA=$(git archive --format=tar --prefix=${PREFIX}/ ${TAG} | gzip | shasum -a 256 | awk '{print $1}')
+
+cat << EOF
+WORKSPACE snippet:
+\`\`\`starlark
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+http_archive(
+    name = "aspect_rules_ts",
```
Comment (alexeagle) [inline .github/workflows/workspace_snippet.sh:17]: copy-pasta needs fix

File: WORKSPACE [hunk 3]
```diff
@@ -9,27 +9,38 @@ load("//:internal_deps.bzl", "rules_webpack_internal_deps")
...
-    symlink_node_modules = False,
+    pnpm_lock = "//webpack/tests:pnpm-lock.yaml",
+)
+
+load("@npm//:repositories.bzl", "npm_repositories")
+
+npm_repositories()
+
+
+load("//webpack:repositories.bzl", "webpack_register_toolchains", "LATEST_WEBPACK_VERSION")
```
Comment (alexeagle) [inline WORKSPACE:39]: everywhere else we just name this symbol `LATEST_VERSION` - I think we should be consistent? (might mean changing all the others, though I typically rename on import like `load(webpack_version = "LATEST_VERSION)`

File: webpack/private/webpack_bundle.bzl [hunk 4]
```diff
@@ -112,10 +111,11 @@ def _desugar_entry_points(name, entry_point, entry_points, inputs):
         result[f[0]] = name
     return result
 
-def _no_ext(f):
-    return f.short_path[:-len(f.extension) - 1]
 
-def _webpack_outs(name, entry_point, entry_points, output_dir):
+def _filter_js(files):
+    return [f for f in files if f.extension == "js" or f.extension == "mjs"]
```
Comment (alexeagle) [inline webpack/private/webpack_bundle.bzl:116]: cjs?

File: webpack/repositories.bzl [hunk 5]
```diff
@@ -1,32 +1,76 @@
...
+load("@aspect_bazel_lib//lib:directory_path.bzl", "directory_path")
+load("@aspect_rules_js//js:defs.bzl", "js_binary")
+load("@aspect_rules_webpack//webpack/private:{version}/defs.bzl", "npm_link_all_packages")
+npm_link_all_packages(name = "node_modules")
+
+directory_path(
+    name = "entrypoint",
+    directory = ":node_modules/webpack/dir",
+    path = "bin/webpack.js",
+    visibility = ["//visibility:public"],
```
Comment (alexeagle) [inline webpack/repositories.bzl:24]: nit: should just set `package(default_visibility)`

**Review by alexeagle — APPROVED** [review]
really close, nice :)

File: WORKSPACE (hunk 3 above)
Comment (thesayyn) [reply WORKSPACE:39]: Agreed. will change

File: webpack/private/webpack_bundle.bzl (hunk 4 above)
Comment (thesayyn) [reply webpack/private/webpack_bundle.bzl:116]: Ah yes. I think we need `.jsx` as well.

**Pull request merged**